### PR TITLE
CDAP-4287: CLI for ACL management

### PR DESCRIPTION
**Pull request by jwang47**
https://issues.cask.co/browse/CDAP-4287
http://builds.cask.co/browse/CDAP-DUT3265

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/security/RevokeActionCommand.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    Set<Action> actions = fromStrings(Splitter.on(",").split(arguments.get("actions")));
+
+    client.revoke(new RevokeRequest(entity, user, actions));
+    output.printf("Successfully revoked action(s) '%s' on entity '%s' for user '%s'\n",
+                  Joiner.on(",").join(actions), entity.toString(), user);
+  }
+
+  @Override
+  public String getPattern() {
+    return "security revoke entity <entity> user <user> actions <actions>";
```
Comment (gokulavasan) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/security/RevokeActionCommand.java]: Isn't ``actions`` an optional argument?

File: cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java [hunk 2]
```diff
@@ -82,6 +82,25 @@ public final EntityType getEntity() {
     return (T) id.toId();
   }
 
+  public static <T extends EntityId> T fromString(String string) {
```
Comment (gokulavasan) [inline cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java]: Will EntityId have autocomplete in CLI?

Comment (gokulavasan) [inline cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java]: What about Actions?

**Comment by gokulavasan** [comment]
Few comments @jwang47

File: cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java (hunk 2 above)
Comment (jwang47) [inline cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java]: No autocompletion right now. We need to add autocompletion for list-style arguments and EntityId.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/security/RevokeActionCommand.java (hunk 1 above)
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/security/RevokeActionCommand.java]: Yeah, will fix.

**Comment by gokulavasan** [comment]
LGTM :+1:

**Pull request merged**